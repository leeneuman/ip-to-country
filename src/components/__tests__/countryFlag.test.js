import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CountryFlag from '../countryFlag';

afterEach(cleanup);

const country = {
	countryCode: 'IL',
	countryCode3: 'ISR',
	countryName: 'Israel',
	countryEmoji: 'ðŸ‡®ðŸ‡±'
};

test('should render flag component', () => {
	render(<CountryFlag country={country} />);
	const flagElement = screen.getByTestId('flag');
	expect(flagElement).toBeInTheDocument();
});

test('matches snapshot', () => {
	const tree = renderer.create(<CountryFlag country={country} />).toJSON();
	expect(tree).toMatchSnapshot();
});
