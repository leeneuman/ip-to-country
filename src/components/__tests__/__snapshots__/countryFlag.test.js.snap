// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="flag"
>
  <img
    aria-label="Israel"
    className="mr-3"
    src="https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.3/flags/4x3/il.svg"
    style={
      Object {
        "display": "inline-block",
        "height": "2em",
        "verticalAlign": "middle",
        "width": "2em",
      }
    }
  />
</div>
`;
